CMakeLists.txt

```diff
@@ -136,6 +136,7 @@ zephyr_compile_options(
   -imacros ${AUTOCONF_H}
```

Comment:
The commit message needs to be re-written.

It should be clear to the reader that this is a
breaking change and how to identify whether he is affected.

The discussion leading up to this change had a good explanation
I believe.

This is so that, when this breaks something, it is easier to identify
where it broke.

Reply:
@SebastianBoe yep make it a bit more verbose. Please  re-review.